spec/logidze/model_spec.rb

```diff
@@ -280,4 +280,76 @@
       expect(user.at(time(250)).log_data.responsible_id).to eq 1
     end
   end
+
+  describe "Versioned associations" do
+    let(:user) do
+      User.create(
+        name: 'John Doe',
+        age: 35,
+        active: true,
+        log_data: {
+          'v' => 3,
+          'h' =>
+            [
+              { 'v' => 1, 'ts' => time(50), 'c' => { 'name' => 'John Harris', 'active' => false, 'age' => 45 } },
+              { 'v' => 2, 'ts' => time(150), 'c' => { 'active' => true, 'age' => 35 } },
+              { 'v' => 3, 'ts' => time(300), 'c' => { 'name' => 'John Doe' } },
+            ]
+        }
+      )
+    end
+
+    let(:post) do
+      Post.create(
+        title: 'Post',
+        rating: 5,
+        active: true,
+        user: user,
+        log_data: {
+          'v' => 3,
+          'h' =>
+            [
+              { 'v' => 1, 'ts' => time(100), 'c' => { 'title' => 'Cool post', 'active' => false } },
+              { 'v' => 2, 'ts' => time(200), 'c' => { 'rating' => 5 } },
+              { 'v' => 3, 'ts' => time(300), 'c' => { 'title' => 'Post' } },
+            ]
+        }
+      )
+    end
+
+    context "with belongs to" do
+      it "returns association version, according to the owner" do
+        post.user
+        old_post = post.at(time(200))
+        expect(old_post.user.name).to eql('John Harris')
+
+        very_old_post = post.at(time(100))
+        expect(very_old_post.user.age).to eql(45)
+      end
+    end
+
+    context "with has many" do
+      before(:each) do
+        post.comments.create(
+          content: 'New comment',
+          log_data: {
+            'v' => 2,
+            'h' =>
+              [
+                { 'v' => 1, 'ts' => time(150), 'c' => { 'content' => 'My comment' } },
+                { 'v' => 2, 'ts' => time(250), 'c' => { 'content' => 'New comment' } },
+              ]
+          }
+        )
+      end
+
+      it "returns association version, according to the owner" do
+        old_post = post.at(time(200))
+        expect(old_post.comments.first.content).to eql('My comment')
+
+        very_old_post = post.at(time(100))
+        expect(very_old_post.comments.length).to eql(0)
```

Comment:
Note the `length` method usage instead of `size`. It corresponds to the problem I described in the issue #19.
Unlike `size` `length` method use `load_target`, so we can get, what we want, here